frontend/frame/TextContentFactory.ts

```diff
@@ -0,0 +1,25 @@
+import { TextContent } from '../../model/slide/content/TextContent';
+import { ContentFactory } from './ContentFactory';
+
+/**
+ * Class which transforms JSON to TextContent Object
+ */
+export class TextContentFactory extends ContentFactory {
+  /**
+   * @inheritDoc
+   * @param json the JSON
+   * @returns the textContent as TextContent type
+   * @protected
+   */
+  public buildContent(json: unknown): TextContent {
+    const textContent = new TextContent();
+    textContent.alignmentHorizontal = (json as TextContent).alignmentHorizontal;
+    textContent.alignmentVertical = (json as TextContent).alignmentVertical;
+    textContent.textSnippets = [];
+
+    (json as TextContent).textSnippets.forEach((snippet) => {
```

Comment:
is this ass really needed? we should be able to iterate over the json object shouldnt we?